File: nagasenacs/src/Nagasena/Proc/IO/Compression/ScriberChannel.cs
using System.Collections.Generic;

using Channel = Nagasena.Proc.Common.Channel;

namespace Nagasena.Proc.IO.Compression {

  internal class ScriberChannel : Channel {

    internal readonly List<ScriberValueHolder> values;

    internal ScriberChannel(int firstPos, int blockNum) : base(firstPos, blockNum) {
      values = new List<ScriberValueHolder>();
    }

    public override void reset(int firstPos, int blockNum) {
      base.reset(firstPos, blockNum);
      values.Clear();
    }

  }

}

File: sumerogi/src/com/sumerogi/proc/io/compression/ScriberChannel.java
package com.sumerogi.proc.io.compression;

import java.util.ArrayList;

import com.sumerogi.proc.common.Channel;

class ScriberChannel extends Channel {

  final ArrayList<ScriberValueHolder> values;
  
  ScriberChannel(int firstPos, int blockNum) {
    super(firstPos, blockNum);
    values = new ArrayList<ScriberValueHolder>();
  }

  @Override
  public void reset(int firstPos, int blockNum) {
    super.reset(firstPos, blockNum);
    values.clear();
  }
  
}
